tests/bdd/zfs.feature

```diff
@@ -0,0 +1,87 @@
+         Validate snapshot creation and clone
+         Validate volume resize
+
+  Background:
+    Given a single zfs cluster is configured
+    And zfs pool pool created on the node
+
+  Scenario: volume provision for file system type zfs, ext4, xfs, btrfs
+    Given a storage class is created with fsType as zfs, ext4, xfs, btrfs respectively 
+    And pvc is created referencing the same sc
+    When a deployment is created using the same pvc
+    Then zfsvolume must be created and used by the application
+
+  Scenario: volume property change for file system type zfs, ext4, xfs, btrfs
+    Given a storage class is created with fsType as zfs, ext4, xfs, btrfs respectively 
+    And pvc is created referencing the same sc
+    And a deployment is created using the same pvc
+    When the zfsvolume properties like compression, dedep are updated
+    Then the zfsvolume properties must reflect the updated value
+
+  Scenario: volume property change for file system type zfs, ext4, xfs
+    Given a storage class is created with fsType as zfs, ext4, xfs, btrfs respectively 
+    And pvc is created referencing the same sc
+    And a deployment is created using the same pvc
+    When the zfsvolume volume type is DATASET and record size is updated
+    Then the record size of must show the updated value of zfsvolume
+
+  Scenario: volume property change for file system type zfs, ext4, xfs
+    Given a storage class is created with fsType as zfs, ext4, xfs, btrfs respectively 
+    And pvc is created referencing the same sc
+    And a deployment is created using the same pvc
+    When the zfsvolume volume type is not DATASET and block size is updated
+    Then the block size of zfsvolume must be updated
+
+  Scenario: create a snaphot from a pvc
+    Given a storage class is created with fsType as zfs, ext4, xfs, btrfs separately 
+    And pvc is created referencing the same sc
+    And a deployment is created using the same pvc
+    And a zfsvolume is created and used by the application
+    When the snapshot create command for the given pvc is run
+    Then the snapshot must be created for the corresponding pvc
+
+  Scenario: resize a pvc when fsType is zfs, ext4, xfs
+    Given a storage class is created with  zfs, ext4, xfs separately 
+    And pvc is created referencing the same sc
+    And a deployment is created using the same pvc
+    When the size of pvc is updated 
+    Then the pvc must relfect the new size
+
+  Scenario: create a clone from a snaphot
+    Given a snaphot for a pvc
+    When the clone create command is run for the given snaphot 
+    Then the clone must be created from the snapshot
+
+  Scenario: create a clone  from a volume
+    Given a pvc
+    When the clone create command is run for the given volume 
+    Then the clone must be created from the volume
+
+  Scenario: volume provision for a raw block volume
+    Given a storage class is created without any fstype as Raw block volume does not have any fstype
```

Comment:
raw block specification is not a part of sc params. It comes from pvc. Please update it accordingly.

Reply:
Update other cases as well